src/serializers/decoders/current/v9/decode_graph_entities.c

```diff
@@ -54,7 +54,9 @@ static SIValue _RdbLoadSIArray(RedisModuleIO *rdb) {
 	uint arrayLen = RedisModule_LoadUnsigned(rdb);
 	SIValue list = SI_Array(arrayLen);
 	for(uint i = 0; i < arrayLen; i++) {
-		SIArray_Append(&list, _RdbLoadSIValue(rdb));
+		SIValue elem = _RdbLoadSIValue(rdb);
+		SIArray_Append(&list, elem);
+		SIValue_Free(elem);
```

Comment:
Right, array need to protect its elements by calling `SI_CloneValue` internally, it's easy to forget that, which will lead to leaks.

Reply:
I feel like we should try switching to a reference counter for `SIValue`s soon; it will reduce the number of redundant allocs greatly.

Reply:
Agree, we need to consider this option